index.ts

```diff
@@ -7,12 +7,10 @@ interface SafeProxy<Initial, To> {
   $pmap: (f: (t: To) => Promise<To>) => Promise<Initial>;
 }
 
-type UnionToIntersection<U> = (U extends any ? (k: U) => void : never) extends (k: infer I) => void
-  ? I
-  : never;
+type KeysOfUnion<T> = T extends any ? keyof T : never;
```

Comment:
By using conditional types TypeScript will go through all types that make up `T` and return their keys 😬

Reply:
...right. might be nice to have some @ts-expect-error tests for things so that we do not end up accepting everything.

Reply:
What do we not want to accept?

Reply:
like `const a: { a: number}: safe(a).b.$` should be an error and stuff like that. Also maybe check that we follow the field types when the type is an union. Or that the resulting type is what you would expect.